plugins/otter-pro/inc/plugins/class-block-conditions.php

```diff
@@ -566,6 +596,122 @@ public function has_products( $products ) {
 		return $bought;
 	}
 
+	/**
+	 * Check based on WooCommerce product category.
+	 *
+	 * @param array $categories IDs of Categories.
+	 *
+	 * @since  2.7.0
+	 * @access public
+	 */
+	public function has_product_category( $categories ) {
+		$in_category = false;
+
+		global $product;
+
+		if ( ! $product ) {
+			return false;
+		}
+
+		$terms = get_the_terms( $product->get_id(), 'product_cat' );
+
+		if ( gettype( $terms ) !== 'array' ) {
+			return false;
+		}
+
+		foreach ( $terms as $term ) {
+			if ( in_array( $term->term_id, $categories, true ) ) {
+				$in_category = true;
+				break;
+			}
+		}
+
+		return $in_category;
+	}
+
+	/**
+	 * Check based on WooCommerce product tags.
+	 * 
+	 * @param array $tags IDs of Tags.
+	 * 
+	 * @since  2.7.0
+	 * @access public
+	 */
+	public function has_product_tag( $tags ) {
+		$in_tag = false;
+
+		global $product;
+
+		if ( ! $product ) {
+			return false;
+		}
+
+		$terms = get_the_terms( $product->get_id(), 'product_tag' );
+
+		if ( gettype( $terms ) !== 'array' ) {
+			return false;
+		}
+
+		foreach ( $terms as $term ) {
+			if ( in_array( $term->term_id, $tags, true ) ) {
+				$in_tag = true;
+				break;
+			}
+		}
+
+		return $in_tag;
+	}
+
+	/**
+	 * Check based on WooCommerce product attribute.
+	 *
+	 * @param array $condition Condition.
+	 *
+	 * @since  2.7.0
+	 * @access public
+	 */
+	public function has_product_attribute( $condition ) {
+		$attribute     = $condition['attribute'];
+		$terms         = isset( $condition['terms'] ) ? $condition['terms'] : false;
+		$has_attribute = false;
+		$in_attribute  = false;
+
+		global $product;
+
+		if ( ! $product ) {
+			return false;
+		}
+
+		$attributes = $product->get_attributes();
+
+		foreach ( $attributes as $attr ) {
+			if ( $attr->get_id() === (int) $attribute ) {
+				$has_attribute = true;
+				$attribute     = $attr;
+				break;
+			}
+		}
+
+		if ( ! $has_attribute ) {
+			return false;
+		}
+
+		if ( ! $terms ) {
+			return true;
+		}
+
+		$attribute_terms = $attribute->get_terms();
```

Comment:
You expect `attribute` to be an object where you use the `get_terms` method. You should be checking if it's an object of the right class and abort if it isn't.

Reply:
This one I don't understand fully. Do you mean to check if get_terms is a method of WC_Product_Attribute class? The returned terms is regular WP taxonomy. Let me know if I understood it wrong.

Reply:
Yep, that's what I want to check! 😃